Dockerfile

```
@@ -0,0 +1,34 @@
+FROM node:14.17.4 AS verify-format
+WORKDIR /src
+COPY package.json yarn.lock ./
+RUN yarn
+COPY . .
+RUN yarn verify-format
+
+FROM koalaman/shellcheck-alpine as verify-sh
+WORKDIR /src
+COPY ./*.sh ./
+RUN shellcheck -e SC1091,SC1090 ./*.sh
+
+FROM node:14.17.4 as restore
+WORKDIR /work
+RUN npm install
+COPY package.json package-lock.json ./
+RUN ls
+RUN npm install --unsafe-perm
```

Creo que se puede hacer que `verify-format` aproveche lo que se hizo en `restore` (cuidado, cambié el orden, habría que cambiar el orden de los llamados en Jenkinsfile)

Por otro lado, cuidado porque estás mezclando el uso de `yarm` y `npm`.

```suggestion
FROM node:14.17.4 AS restore
WORKDIR /src
COPY package.json yarn.lock ./
RUN yarn
COPY . .

FROM restore AS verify-format
RUN yarn verify-format

FROM koalaman/shellcheck-alpine as verify-sh
WORKDIR /src
COPY ./*.sh ./
RUN shellcheck -e SC1091,SC1090 ./*.sh
```

---

Cool. Changed.